tests/PyomoTest.py

```diff
@@ -0,0 +1,283 @@
+  cfParams = {'name': 'Hourly',
+               'X': 1,
+               'mult_target': None,
+               'inflation': False}
+  cf.setParams(cfParams)
+  cf.initParams(life, pyomoVar=True)
+  for year in range(life):
+    if isinstance(alpha, float):
+      cf.computeIntrayearCashflow(year, alpha, driver[year, :])
+    else:
+      cf.computeIntrayearCashflow(year, alpha[year, :], driver[year, :])
+  return cf
+
+
+# main
+if __name__ == '__main__':
+  # problem characteristics
+  project_life = 5
+  hours_in_year = 10
+  hours = np.arange(hours_in_year)
+
+  # create the global cashflow settings
+  expected_cfs = {'Generator': ['Cap', 'FixedOM', 'Hourly'],
+                  'Market': ['Hourly']}
+  tealSettings = build_econ_settings(expected_cfs, life=project_life)
+
+  m = pyo.ConcreteModel()
+  m.T = pyo.Set(initialize=hours)
+
+  ## for this problem, let's just have 2 components, a source and a sink
+  ## - source ("generator")
+  ##   - has a capex, fixed OM (yearly recurring), and variable OM (hourly recurring)
+  ##   - variable total size
+  ##   - variable dispatch
+  ## - sink ("market")
+  ##   - has sales profit (hourly recurring)
+  ##   - fixed size
+
+  # then we start creating components
+  tealComps = {}
+
+
+  # *** GENERATOR ***
+  # make the Pyomo variables, constants
+  generator_life = 5 # years of operation before replacement
+  generator_size = pyo.Var(initialize=1, bounds=(0, 100)) # build size, limit to 100 MW
+  m.gen_size = generator_size
+
+  # set up the generator's dispatch variables
+  generator_dispatch = np.zeros((generator_life, hours_in_year), dtype=object)
+  for y in range(generator_life):
+    # make a new pyomo var for this "year"'s dispatch
+    var = pyo.Var(list(range(hours_in_year)), initialize=lambda m, t: 0, bounds=lambda m, t: (0, 100)) #generator_size))
+    setattr(m, f'Gen_disp_year_{y+1}', var)
+    generator_dispatch[y, :] = np.array(list(var.values()))
+
+  generator = build_generator(generator_size, generator_life, generator_dispatch)
+
+  # *** MARKET ***
+  # make the TEAL component
+  market_size = 50 # market depth
+  market_life = 5 # years of operation before replacement
+  # market prices should be interesting, say cos(x) + y/10 where x is hour and y is year
+  prices = np.ones((market_life, hours_in_year))
+  prices[:] *= np.cos(hours) * 100
+  for y in range(market_life):
+    if y % 2 == 0:
+      prices[y, :] *= -1
+
+  # market has similar dispatch to generator, that we'll attach through constraints
+  market_dispatch = np.zeros((market_life, hours_in_year), dtype=object)
+  for y in range(market_life):
+    # make a new pyomo var for this "year"'s dispatch
+    var = pyo.Var(list(range(hours_in_year)), initialize=lambda m, t: 0, bounds=lambda m, t: (0, market_size))
+    setattr(m, f'Market_disp_year_{y+1}', var)
+    market_dispatch[y, :] = np.array(list(var.values()))
+
+  market = build_market(market_size, market_life, prices, market_dispatch)
+
+  # DEBUGG
```

Comment:
this commented line can be removed, it was a placeholder

Reply:
Done